generator/test_case_generator.go

```diff
@@ -76,6 +76,13 @@ var testTypeToTestDirMap = map[string][]string{
 	"ec2_acceptance": {
 		"./test/acceptance",
 	},
+	"ec2_userdata": {
+		"./test/cloudwatchlogs",
+		"./test/metric_value_benchmark",
+		"./test/run_as_user",
+		"./test/userdata",
```

Comment:
The only difference is userdata pushes metric to a namespace called UserDataTest. This is to distinguish userdata CWAgent test by verifying that CWAgent can in fact start on launch using userdata. If just using the metric_value_benchmark it can get mixed up with the post-launch behavior instead since they will be under same namespace.

Reply:
But when the "./test/metric_value_benchmark" suite is being run, none of the tests in that suite match your `Userdata` if condition logic right?